### PR TITLE
[TOOLS-1770] Basic Cluster CRUD - Backend

### PR DESCRIPTION
**Pull request by JuniorMiqueletti**

### CODE REVIEW

File: src/main/java/objective/taskboard/cluster/base/BaseClusterService.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    private Optional<SizingClusterItem> findCorrespondingClusterItem(final SizingClusterItem item, final List<SizingClusterItem> savedClsuterItems) {
+        if (savedClsuterItems.isEmpty())
+            return Optional.empty();
+
+        return savedClsuterItems.stream()
+                .filter(i -> matchesClusterItem(item, i.getSubtaskTypeName(), i.getParentTypeName(), i.getSizing()))
+                .findFirst();
+    }
+
+    private Boolean matchesClusterItem(final SizingClusterItem item, final String subtaskTypeName, final String parentTypeName,
```
Comment (objective-solutions-user) [inline src/main/java/objective/taskboard/cluster/base/BaseClusterService.java]: MAJOR Remove this unused method parameter "parentTypeName". [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS1172)

File: src/main/java/objective/taskboard/cluster/base/BaseClusterService.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            final String sizing) {
+        return item.getSubtaskTypeName().equals(subtaskTypeName) && item.getSubtaskTypeName().equals(subtaskTypeName) && item.getSizing().equals(sizing);
```
Comment (objective-solutions-user) [inline src/main/java/objective/taskboard/cluster/base/BaseClusterService.java]: MAJOR Correct one of the identical sub-expressions on both sides of operator "&&" [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS1764)

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 2 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: src/main/java/objective/taskboard/cluster/ClusterRepository.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package objective.taskboard.cluster;
+
+import java.util.List;
+import java.util.Optional;
+
+import objective.taskboard.followup.cluster.SizingCluster;
+
+public interface ClusterRepository {
+    Optional<SizingCluster> findById(Optional<Long> cluster);
```
Comment (jytakano) [inline src/main/java/objective/taskboard/cluster/ClusterRepository.java]: Why two methods `findById`?

**Pull request merged**